### PR TITLE
chore(workflow): add test for activity retry policies

### PR DESCRIPTION
**Pull request by mjameswh**
## What changed

- Added an integration test to assert fix of #441

### CODE REVIEW

File: packages/test/src/test-integration-workflows.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ import {
 } from '@temporalio/worker';
 import * as activity from '@temporalio/activity';
 import * as workflow from '@temporalio/workflow';
-import { CancelReason } from '@temporalio/worker/src/activity';
+import { CancelReason } from '@temporalio/worker/lib/activity';
```
Comment (bergundy) [inline packages/test/src/test-integration-workflows.ts:17]: We should export this from worker or activity if we don’t already

**Review by bergundy — APPROVED** [review]

File: packages/test/src/test-integration-workflows.ts (hunk 1 above)
Comment (mjameswh) [reply packages/test/src/test-integration-workflows.ts:17]: ATM, that would have to be exported from 'worker'...

But why? We don't use that type on anything public, except for `CancelledFailure.message`, but we can't easily change the type of the `message` property (it's in common and may also be used in the context of a Workflow, which implies different set of values).

Comment (bergundy) [reply packages/test/src/test-integration-workflows.ts:17]: Still useful for users to compare and I'd like to have a property on activity context for the cancel reason.

Comment (mjameswh) [reply packages/test/src/test-integration-workflows.ts:17]: Another thing... on deserializing a `CancelledFailure` object from an incoming failure (ie. failure-converter.ts), the incoming message may come from a different SDK, and could therefore contain any message (ie. not bound to `CancelReason`)

Comment (bergundy) [reply packages/test/src/test-integration-workflows.ts:17]: We don't have any API that throws CancelledFailure directly.

Comment (bergundy) [reply packages/test/src/test-integration-workflows.ts:17]: I don't think the message property should be typed, we use CancelledFailure all over the SDK.

**Pull request merged**